pandas/src/algos_groupby_helper.pxi.in

```diff
@@ -568,6 +568,76 @@ def group_min_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
                 else:
                     out[i, j] = minx[i, j]
 
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def group_cummin_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
+                          ndarray[{{dest_type2}}, ndim=2] values,
+                          ndarray[int64_t] labels,
+                          ndarray[{{dest_type2}}, ndim=2] accum):
```

Comment:
@jreback - I had done this pattern in the `cumsum` code.  IIRC, it was because I was using fused types / memorviews, and I didn't see a way to do the allocation.  It would be straightforward to avoid in the templated code.

By example
```
def func(numeric[:] arr, int a, int b):
    # how would you allocate an (a x b) array of type `numeric?`
```